test/Microsoft.ML.Tests/TextLoaderTests.cs

```diff
@@ -7,21 +7,147 @@
+using System.IO;
 using Xunit;
 using Xunit.Abstractions;
 
 namespace Microsoft.ML.EntryPoints.Tests
 {
+    public class TextLoaderTestPipe : TestDataPipeBase
+    {
+        public TextLoaderTestPipe(ITestOutputHelper output)
+            : base(output)
+        {
+
+        }
+
+        [Fact]
+        public void TestTextLoaderDataTypes()
+        {
+            string pathData = DeleteOutputPath("SavePipe", "TextInput.txt");
+            File.WriteAllLines(pathData, new string[] {
+                string.Format("{0},{1},{2},{3}", sbyte.MinValue, short.MinValue, int.MinValue, long.MinValue),
+                string.Format("{0},{1},{2},{3}", sbyte.MaxValue, short.MaxValue, int.MaxValue, long.MaxValue),
+                ",,,"
+            });
+
+            try
+            {
+                var data = TestCore(pathData, true,
+                    new[] {
+                    "loader=Text{col=DvInt1:I1:0 col=DvInt2:I2:1 col=DvInt4:I4:2 col=DvInt8:I8:3 sep=comma}",
+                    }, logCurs: true);
+
+                using (var cursor = data.GetRowCursor((a => true)))
+                {
+                    var col1 = cursor.GetGetter<sbyte>(0);
+                    var col2 = cursor.GetGetter<short>(1);
+                    var col3 = cursor.GetGetter<int>(2);
+                    var col4 = cursor.GetGetter<long>(3);
+
+                    Assert.True(cursor.MoveNext());
+
+                    sbyte[] sByteTargets = new sbyte[] { sbyte.MinValue, sbyte.MaxValue, default};
+                    short[] shortTargets = new short[] { short.MinValue, short.MaxValue, default };
+                    int[] intTargets = new int[] { int.MinValue, int.MaxValue, default };
+                    long[] longTargets = new long[] { long.MinValue, long.MaxValue, default };
+
+                    int i = 0;
+                    for (; i < sByteTargets.Length; i++)
+                    {
+                        sbyte sbyteValue = -1;
+                        col1(ref sbyteValue);
+                        Assert.Equal(sByteTargets[i], sbyteValue);
+
+                        short shortValue = -1;
+                        col2(ref shortValue);
+                        Assert.Equal(shortTargets[i], shortValue);
+
+                        int intValue = -1;
+                        col3(ref intValue);
+                        Assert.Equal(intTargets[i], intValue);
+
+                        long longValue = -1;
+                        col4(ref longValue);
+                        Assert.Equal(longTargets[i], longValue);
+
+                        if (i < sByteTargets.Length - 1)
+                            Assert.True(cursor.MoveNext());
+                        else
+                            Assert.False(cursor.MoveNext());
+                    }
+
+                    Assert.Equal(i, sByteTargets.Length);
+                }
+            }
+            catch
+            {
+                Assert.True(false, "Test failed.");
+            }
```

Comment:
If an exception gets thrown in the test, that will fail the test. But we'll also get to see what the exception is. By concealing it in this `catch` we hide the exception, which makes the test needlessly complex (since this isn't a test where we expect exceptions) and makes the result harder to interpret.